frontend/src/components/common/Tab.tsx

```diff
@@ -20,25 +20,42 @@ interface TabsProps {
   tabProps?: {
     [propName: string]: any;
   };
+  defaultIndex?: number | null | boolean;
   onTabChanged?: (tabIndex: number) => void;
 }
 
 export default function Tabs(props: TabsProps) {
-  const {tabs, tabProps = {}, onTabChanged = null} = props;
-  const [tabIndex, setTabIndex] = React.useState(0);
+  const {
+    tabs,
+    tabProps = {},
+    defaultIndex = 0,
+    onTabChanged = null
+  } = props;
+  const [tabValue, setTabValue] = React.useState<TabsProps['defaultIndex']>(
+    defaultIndex && Math.min(defaultIndex as number, 0));
 
   function handleTabChange(event: any, newValue: number) {
-    setTabIndex(newValue);
+    setTabValue(newValue);
```

Comment:
There's no need for this because what we're setting is the index and thus changing the name to value makes the code less clear IMO.